### ISSUE TITLE
Error should be reseted in every request.

### ISSUE THREAD

**Issue by airamrguez** [alex-cory/use-http#98]
Hi!

`useHttp` is really easy and cool to use. Congrats!
I would like to expose an issue I'm experimenting. I have something like this:

```js
function SomeContainer() {
  const { error, request } = useFetch('https://...');
  const [data, setData] = useState(null);
  const [query, setQuery] = useState({ page: 1 });

  useEffect(() => {
    // ... A wrapper to allow using async/await
    const serializedQuery = queryString.stringify(query);
    const data = await request.get(`/some/endpoint/${serializedQuery}`);
    setData(data);
  }, [request, query]);

  const onQueryChange = nextQuery => {
    setQuery(nextQuery);
  };

  if (error) {
    return 'Something went wrong';
  }
  if (loading) {
    return 'Loading ...';
  }
  return <SomeDataComponent data={data}  onQueryChange={onQueryChange} />
}
```
Suppose that:
1) The user has internet connection. Everything works fine. The result is that 'Loading ...' text appears first and then `SomeDataComponent` is rendered.
2) The user lose connection.
3) The user modifies the query and `onQueryChange` is called causing a re-render. That re-render executes the `useEffect` that will make a new fetch. The result of that fetch will be an error and we will see the text 'Something went wrong' in the screen.
4) The user is online again.
5) The user modify the query again so a new re-fetch is performed. In this scenario, in which the internet is reachable, we won't see `SomeDataComponent` rendered again. Instead we will stuck on 'Something went wrong'.

So to sum up: the `error` state is not reseted across requests calls.

**Comment by alex-cory** [comment]
I will take a look at this today! :)

**Comment by alex-cory** [comment]
This should be fixed in [this PR](https://github.com/alex-cory/use-http/pull/100/files). Merging once checks are completed

**Comment by alex-cory** [comment]
This has been tested and published. Closing this. If the issue persists, feel free to re-open this issue.

**Comment by airamrguez** [comment]
Cool! Thanks @alex-cory
